src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4228,21 +4226,9 @@ private string GetSimpleStaticResourceRetrieval(INamedTypeSymbol? targetProperty
 			targetPropertyType = targetPropertyType ?? _objectSymbol;
 
 			var targetPropertyFQT = targetPropertyType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
-			var propertyOutputVar = $"staticResourceResolverOutputIndex{_staticResourceResolverOutputIndex++}";
-
-			var conversion = targetPropertyType.IsValueType
-				? $"(({targetPropertyFQT}){propertyOutputVar})"
-				: $"(({propertyOutputVar} as {targetPropertyFQT}) ?? default({targetPropertyFQT}))";
-
 
-			var staticRetrieval = $"(" +
-				$"global::Uno.UI.ResourceResolverSingleton.Instance.ResolveResourceStatic(" +
-				$"\"{keyStr}\", " +
-				$"out var {propertyOutputVar}, " +
-				$"context: {ParseContextPropertyAccess}) " +
-				$"? {conversion}" +
-				$" : default({targetPropertyFQT})" +
-				$")";
+			var staticRetrieval = $"global::Uno.UI.ResourceResolverSingleton.Instance.ResolveResourceStatic<{targetPropertyFQT}>(" +
```

Comment:
@jeromelaban is it ok to use generics here? I seem to recall there were some size-related issues in generated code.

Reply:
@davidjohnoliver It's used in other places:

```
"new global::Windows.UI.Xaml.Setter<{0}>(\"{1}\", o => {2}.{1} = "
```

Reply:
If it's a problem, it would be good to clean other occurrences. Side note: Generics type inference may be used and can be hard to spot. So we need some mechanism to easily detect that if it's indeed problematic?

Reply:
It's generally best to avoid generics in those cases indeed, particularly if we only use a `typeof` afterwards. As for an analyzer, not at this point.